### PR TITLE
Use staged macro

### PR DESCRIPTION
**Pull request by aalkin**
A new macro was added to O2 (https://github.com/AliceO2Group/AliceO2/pull/13624) to simplify declaration of a Table/StoredTable pair. Here I've implemented the change. Note that this is a transparent change, the processing is not affected, only the declaration is rearranged.

### CODE REVIEW

**Review by nzardosh — APPROVED** [review]

**Comment by nzardosh** [comment]
Hi @aalkin , I approved these formatting changes but do you really want to make them?

**Review by njacazio — APPROVED** [review]
Super, thanks a lot @aalkin

**Comment by aalkin** [comment]
Hi @nzardosh, combining the two declarations into one is needed for the upcoming upgrade in O2 where declaring the separate tables with the same description would not work. Additionally, the pattern Table(AOD)/StoredTable(AOD1) is used consistently, so it made sense to have it declared with a single macro. If you have any suggestions or use cases where this is not enough or not optimal - we can always improve.

**Review by vkucera — APPROVED** [review]

**Comment by vkucera** [comment]
Thanks a lot @aalkin !

**Comment by aalkin** [comment]
@iarsene @lucamicheletti93 Could you please review the changes and approve for DQ?

**Review by ddobrigk — APPROVED** [review]

**Review by iarsene — APPROVED** [review]

File: PWGHF/DataModel/DerivedTables.h [hunk 1]
```diff
@@ -51,49 +50,50 @@ DECLARE_SOA_COLUMN(MultZeqNTracksPV, multZeqNTracksPV, float); //! z-equalised b
...
+                         soa::Marker<MarkerD0>);
 
 using HfD0CollBase = HfD0CollBases::iterator;
+using StoredHfD0CollBase = StoredHfD0CollBases::iterator;
 
-DECLARE_SOA_TABLE(HfD0CollIds, "AOD", "HFD0COLLID", //! Table with original global indices of collisions
-                  hf_cand::CollisionId,
-                  soa::Marker<MarkerD0>);
+DECLARE_SOA_TABLE_STAGED(HfD0CollIds, "HFD0COLLID", //! Table with original global indices of collisions
+                         hf_cand::CollisionId);
```
Comment (vkucera) [inline PWGHF/DataModel/DerivedTables.h:72]: @aalkin Was the removal of the marker column intentional?

**Pull request merged**